R/na.R

```diff
@@ -33,22 +35,37 @@
 #' levels(f3)
 #' as.integer(f3)
 #' is.na(f3)
-fct_na_value_to_level <- function(f) {
+fct_na_value_to_level <- function(f, level = NA) {
+  if (!identical(level, NA) && !is_string(level)) {
+    cli::cli_abort(
+      "{.arg level} must be a string or {.code NA}, not {.obj_type_friendly level}."
+    )
+  }
   f <- check_factor(f)
 
-  # re-matches values to levels
-  fct_expand(f, NA)
+  f <- fct_expand(f, NA)
+  new_levels <- levels(f)
+  new_levels[is.na(new_levels)] <- level
+
+  lvls_revalue(f, new_levels)
```

Comment:
So if you already have an `NA` level, this will rename it to `level`? Seems reasonable